contracts/NameHash.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.4.11;
+
+import "github.com/Arachnid/solidity-stringutils/strings.sol";
+
+
+/**
+ * Namehash algorighm implementation in Solidity.
+ */
+library NameHash
+ {
+    using strings for *;
+    
+    
+    /**
+     * Native namehash algorighm implementation in Solidity.
+     * @param  _node The string over which the namehash algorithm will be applied.
+     * @return _namehash The result of namehash algorighm applied at _node.
+     */
+    function namehash(string _node) constant returns (bytes32 _namehash)
+    {
+        var node = _node.toSlice();
+        bytes32 namehash = 0x0000000000000000000000000000000000000000000000000000000000000000;
+        if(!node.empty())
+        {
+            var delim = ".".toSlice();
+            var parts = new string[](node.count(delim) + 1);
+            for(uint i = 0; i < parts.length; i++)
```

Comment:
Perhaps `uint256` instead of `uint`, to prevent any possible compiler misrepresentation?..